hpke/shortkem.go

```diff
@@ -0,0 +1,147 @@
+package hpke
+
+import (
+	"crypto/elliptic"
+	"crypto/rand"
+	_ "crypto/sha256" // linking sha256 packages.
+	_ "crypto/sha512" // linking sha512 packages.
+	"crypto/subtle"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/cloudflare/circl/kem"
+)
+
+type shortKem struct {
+	kemBase
+	elliptic.Curve
+}
+
+func (s shortKem) PrivateKeySize() int        { return s.byteSize() }
+func (s shortKem) SeedSize() int              { return s.byteSize() }
+func (s shortKem) CiphertextSize() int        { return 1 + 2*s.byteSize() }
+func (s shortKem) PublicKeySize() int         { return 1 + 2*s.byteSize() }
+func (s shortKem) EncapsulationSeedSize() int { return s.byteSize() }
+
+func (s shortKem) byteSize() int { return (s.Params().BitSize + 7) / 8 }
+
+func (s shortKem) sizeDH() int { return s.byteSize() }
+func (s shortKem) calcDH(dh []byte, sk kem.PrivateKey, pk kem.PublicKey) error {
+	PK, ok := pk.(*shortPubKey)
+	if !ok {
+		return kem.ErrTypeMismatch
```

Comment:
It might be better to handle this error earlier, e.g., when the Sender and Receiver are constructed.

Reply:
added some verification in earlier steps